docs/rules/prefer-stateless-function.md

```diff
@@ -43,3 +44,14 @@ class Foo extends React.Component {
   }
 }
 ```
+
+The following pattern is not considered warnings:
+
+```js
+class Foo extends React.PureComponent {
```

Comment:
Being a PureComponent does not make it stateless either - the two things are orthogonal. I'm saying `prefer-stateless-function` is far more important than `prefer-optimization`.